src/codewhisperer/service/keyStrokeHandler.ts

```diff
@@ -204,4 +204,39 @@ export class KeyStrokeHandler {
             }
         }
     }
+
+    /**
+     * Logic to determine if the code changes is done by human typing manually
+     * NOTE: we consider intelli sense recommendation is this type
+     * e.g. extension generated code snippet / copy & paseted etc.
+     */
+    isTextChangedHumanTyping(textChanged: string): boolean {
```

Comment:
You can rename `isTextChangedHumanTyping` to `isTextChangeFromUser`

We may trigger CWSPR when there is an intelliSense acceptance from user or user typed some character.

No CWSPR trigger when doing code formatting.

Reply:
yepp, renamed the method and add another layer to prevent triggering by extension like Prettier or other formatters